packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts

```diff
@@ -100,106 +96,162 @@ describe('UserController', () => {
   });
 
   it('throws error for POST /users with a string', async () => {
-    await client
+    const res = await client
       .post('/users')
       .send('hello')
       .expect(415);
+    expect(res.body.error.message).to.equal(
+      'Content-type application/x-www-form-urlencoded does not match [application/json].',
+    );
   });
 
   it('returns a user with given id when GET /users/{id} is invoked', async () => {
     const newUser = await userRepo.create(user);
     delete newUser.password;
     delete newUser.orders;
-    // MongoDB returns an id object we need to convert to string
-    // since the REST API returns a string for the id property.
-    newUser.id = newUser.id.toString();
 
     await client.get(`/users/${newUser.id}`).expect(200, newUser.toJSON());
   });
 
-  describe('user product recommendation (service) api', () => {
-    // tslint:disable-next-line:no-any
-    let recommendationService: HttpServer;
-
-    before(async () => {
-      recommendationService = createRecommendationServer();
-      await recommendationService.start();
-    });
+  describe('authentication', () => {
+    it('login returns a valid token', async () => {
+      await client
+        .post('/users')
+        .send(user)
+        .expect(200);
 
-    after(async () => {
-      await recommendationService.stop();
+      const res = await client
+        .post('/users/login')
+        .send({email: user.email, password: user.password})
+        .expect(200);
+
+      const token = res.body.token;
+      expect(token).to.not.be.Null();
+      expect(token).to.be.String();
+      expect(token).to.not.be.empty();
+      const parts = token.split('.');
+      //token has 3 parts separated by '.'
+      expect(parts.length).to.equal(3);
```

Comment:
If we want to assert the format of the token, can we use a regular expression instead?

Reply:
> If we want to assert the format of the token, can we use a regular expression instead?

As per our conversation offline, I will remove the test for 3 parts of the jwt token string, since this is tested later on when a jwt token is created and then used to go to /users/me endpoint.